impeller/entity/contents/filters/blend_filter_contents.cc

```diff
@@ -109,7 +110,8 @@ static std::optional<Snapshot> AdvancedBlend(
     auto sampler = renderer.GetContext()->GetSamplerLibrary()->GetSampler({});
     FS::BindTextureSamplerDst(cmd, dst_snapshot->texture, sampler);
     blend_info.dst_y_coord_scale = dst_snapshot->texture->GetYCoordScale();
-    blend_info.dst_input_alpha = absorb_opacity ? dst_snapshot->opacity : 1.0f;
+    blend_info.dst_input_alpha =
+        (absorb_opacity ? dst_snapshot->opacity : 1.0) * alpha.value_or(1.0);
```

Comment:
Since this `alpha` value is no longer being used by `AtlasContents`, we could repurpose it to just forward down the filter chain by multiplying it into the returned `Snapshot::opacity`, which would allow us to remove the need for the extra subpass as mentioned in my other comment (this snapshot alpha forwarding can be supported for both the pipeline and advanced blend case with no inconsistencies/ill effects).

Reply:
ahh, Yeah that is what I missed. I had a similar technique for the pipeline blends which didn't work since it would just blend the alpha with one of the inputs. Will fix so both are applied to the snapshot.